File: problem_0029.py
#!/usr/bin/python
"""http://projecteuler.net/problem=29"""

def main():
	terms = []
	for a in range(2,101):
		for b in range(2,101):
			terms.append(a**b)
	print len(list(set(terms)))

if __name__ == "__main__":
	main()

File: problem_0029.rs
#[macro_use]
extern crate clap;

use common::set_log_level;
use std::collections::HashSet;
use num::BigInt;
use num::pow::pow;

fn main() {
    let args = clap_app!(app =>
        (about: "Solve Project Euler Problem 29, https://projecteuler.net/problem=29")
        (@arg verbose: -v +multiple "Increase log level")
        (@arg limit: "limit")
    ).get_matches();
    set_log_level(&args);

    let limit = args.value_of("limit").map(|n| n.parse::<usize>().unwrap()).unwrap_or(100);

    let mut result: HashSet<BigInt> = HashSet::new();
    for a in 2..(limit+1) {
        for b in 2..(limit+1) {
            result.insert(pow(BigInt::from(a), b));
        }
    }
    println!("{}", result.len());
}
